### PR TITLE
Adding the due date on invoices

### PR DESCRIPTION
**Pull request by codevio**
The due date can be editable on each invoice.
The due date can be set to null, so no due date appears on invoice.
The default due date is set to 3 months.

### CODE REVIEW

**Comment by codevio** [comment]
We should update readme to show how to change the due_date.

File: Classes/Invoice.php [hunk 1]
```diff
@@ -129,6 +115,21 @@ class Invoice
      */
     public $footnote;
 
+    /**
+
+     * Invoice Tax Rates Default.
+     *
+     * @var array
+     */
+    public $tax_rates;
```
Comment (ConsoleTVs) [inline Classes/Invoice.php]: What does this have to do with the due date feature?

File: Classes/Invoice.php [hunk 2]
```diff
@@ -149,15 +150,15 @@ public function __construct($name = 'Invoice')
...
-        $this->tax = config('invoices.tax');
-        $this->tax_type = config('invoices.tax_type');
         $this->decimals = config('invoices.decimals');
         $this->logo = config('invoices.logo');
         $this->logo_height = config('invoices.logo_height');
         $this->date = Carbon::now();
         $this->business_details = Collection::make(config('invoices.business_details'));
         $this->customer_details = Collection::make([]);
         $this->footnote = config('invoices.footnote');
+        $this->tax_rates = config('invoices.tax_rates');
```
Comment (ConsoleTVs) [inline Classes/Invoice.php]: What does this have to do with the due date feature?

File: Config/invoices.php [hunk 3]
```diff
@@ -101,4 +76,36 @@
...
+  |
+  | This array group multiple tax rates.
+  |
+  | The tax type accepted values are: 'percentage' and 'fixed'.
+  | The percentage type calculates the tax depending on the invoice price, and
+  | the fixed type simply adds a fixed ammount to the total price.
+  | You can't mix percentage and fixed tax rates.
+  */
+
+  'tax_rates' => [
```
Comment (ConsoleTVs) [inline Config/invoices.php]: What does this have to do with the due date feature?

File: Templates/default.blade.php [hunk 4]
```diff
@@ -15,6 +15,9 @@
             </div>
             <div style="margin-left:300pt;">
                 <b>Date: </b> {{ $invoice->date->formatLocalized('%A %d %B %Y') }}<br />
+                @if ($invoice->due_date)
+                    <b>Due date: </b>{{ $invoice->due_date->formatLocalized('%A %d %B %Y') }}<br />
```
Comment (ConsoleTVs) [inline Templates/default.blade.php]: Looks OK

File: Templates/default.blade.php [hunk 5]
```diff
@@ -97,14 +100,16 @@
...
                         </tr>
-                        <tr>
-                            <td>
-                                <b>
-                                    Taxes {{ $invoice->tax_type == 'percentage' ? '(' . $invoice->tax . '%)' : '' }}
-                                </b>
-                            </td>
-                            <td>{{ $invoice->taxPriceFormatted() }} {{ $invoice->formatCurrency()->symbol }}</td>
-                        </tr>
+                        @foreach($invoice->tax_rates as $tax_rate)
```
Comment (ConsoleTVs) [inline Templates/default.blade.php]: What does this have to do with the due date feature?

File: Traits/Setters.php [hunk 6]
```diff
@@ -211,4 +211,20 @@ public function footnote($footnote)
...
+     * @param Carbon $due_date
+     *
+     * @return self
+     */
+    public function due_date(Carbon $due_date = null)
+    {
+        $this->due_date = $due_date;
+
+        return $this;
+    }
```
Comment (ConsoleTVs) [inline Traits/Setters.php]: Looks great

**Comment by codevio** [comment]
@ConsoleTVs Sorry, there is a bug with my branch.. I'll redo it.

**Comment by ConsoleTVs** [comment]
Not to worry! I'll be happy to accept it once it's fixed!

Thanks for your time
